course/forms.py

```diff
@@ -93,3 +94,35 @@ def __init__(self, *args, **kwargs):
     class Meta:
         model = StudentGroup
         fields = ['members']
+
+
+class EnrollStudentsForm(forms.Form):
+
+    custom_errors = { 'invalid_choice': _('Selected users must be existing students who have not yet enrolled.')}
+    user_profiles = forms.ModelMultipleChoiceField(
+        queryset=UserProfile.objects.none(),
+        required=True,
+        error_messages=custom_errors
+    )
+
+    def __init__(self, *args, **kwargs):
+        self.instance = kwargs.pop('course_instance')
+        super().__init__(*args, **kwargs)
+        self.fields['user_profiles'].widget.attrs['class'] = 'search-select'
+        self.fields['user_profiles'].queryset = UserProfile.objects.exclude(
```

Comment:
This is too heavy. We have around 14k people in the database and loading this widget will take couple of minutes.

Reply:
Using `ModelMultipleChoiceField` is not possible without setting possible choices in the queryset.
We still propably want that multiple students can be added at a time. Is it better to have the students added by student ids? Or can we do something else to have the field suggest the possible values?

Reply:
To my understanding, the "add assistants" field in the edit course page uses the same logic. It includes all users in the HTML.

Reply:
@raphendyr Do you have any suggestions, how we should handle this efficiency problem.

My proposition is that we used some other field which could handle multiple strings. The user would add new students by student ids. Validation would not require that much modification. It would work fast, but the teacher wouldn't be able to see, if the id they had written was valid.